### PR TITLE
Add integration tests

### LINKED ISSUE #71: Test helm installation
Independently of testing the failover controller logic, we should have an automated test that validates that installing the failover controller can be installed and that its pods become _ready_.

### PR DESCRIPTION
**Pull request by alpeb**
Closes #71

Adds the `integration-test.yml` workflow with the `docker-build` and `install-test` jobs.

`docker-build` builds, caches, and uploads as an artifact the `failover` image, very similar as how we do in `linkerd2`.

`install-test`:
  - creates a k3d cluster
  - downloads and loads `failover` into that cluster
  - installs linkerd
  - installs linkerd-smi
  - installs the helm chart under `charts/linkerd-failover`
  - runs a series of tests

These tests use the github action `.github/actions/failover-test` that calls `helm upgrade -i` to install/upgrade the chart under `charts/linkerd-failover-tests`.
This chart sets up podinfo pods with their corresponding services and one TrafficSplit.
This action is called repeatedly with different configs, and each time `helm test` is used to run the pod `chart/linkerd-failover-tests/templates/tests/curl-pod.yaml` which calls the apex service and validates that the expected pods respond.

### CODE REVIEW

File: .github/actions/failover-test/action.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    required: true
+  east:
+    description: podinfo east available
+    required: true
+runs:
+  using: composite
+  steps:
+  - shell: bash
+    run: |
+      helm upgrade -i -n linkerd-failover-tests --create-namespace --wait \
```
Comment (olix0r) [inline .github/actions/failover-test/action.yml:18]: Super minor/tioli, but might be a little clearer to use the full flag name

```suggestion
      helm upgrade --install -n linkerd-failover-tests --create-namespace --wait \
```

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        mkdir -p $ARCHIVES
+        docker save "${{ env.DOCKER_REGISTRY}}/failover:${{ env.DOCKER_TAG }}" > $ARCHIVES/failover.tar
+    # `with.path` values do not support environment variables yet, so an
+    # absolute path is used here.
+    # https://github.com/actions/upload-artifact/issues/8
+    - name: Upload artifact
+      uses: actions/upload-artifact@82c141cc518b40d92cc801eee768e7aafc9c2fa2
+      with:
+        name: image-archives
+        path: /home/runner/archives
```
Comment (olix0r) [inline .github/workflows/integration-test.yml:44]: I'm surprised that you can't have actions do the templating
```suggestion
    - name: Upload artifact
      uses: actions/upload-artifact@82c141cc518b40d92cc801eee768e7aafc9c2fa2
      with:
        name: image-archives
        path: ${{ env.ARCHIVES }}
```

not a big deal...

File: bin/docker-build-failover [hunk 3]
```diff
@@ -9,14 +9,14 @@ fi
 
 bindir=$( cd "${BASH_SOURCE[0]%/*}" && pwd )
 rootdir=$( cd "$bindir"/.. && pwd )
-# shellcheck source=_tag.sh
-. "$bindir"/_tag.sh
```
Comment (olix0r) [inline bin/docker-build-failover:13]: Can we remove the `_tag.sh` script now?

**Review by olix0r — APPROVED** [review]
Huge fan of this!

File: bin/docker-build-failover (hunk 3 above)
Comment (alpeb) [reply bin/docker-build-failover:13]: yep

File: .github/actions/failover-test/action.yml (hunk 1 above)
Comment (alpeb) [reply .github/actions/failover-test/action.yml:18]: :+1:

File: .github/workflows/integration-test.yml (hunk 2 above)
Comment (alpeb) [reply .github/workflows/integration-test.yml]: That was an old limitation. I've just cleaned that up :+1:

File: charts/linkerd-failover-tests/templates/tests/curl-pod.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    image: curlimages/curl:7.78.0
+    command: ["/bin/sh"]
+    args:
+    - -ec
+    - |
+      success_west=false
+      success_central=false
+      success_east=false
+
+      for i in 1 2 3 4 5 6 7 8 9 10; do
```
Comment (adleong) [inline charts/linkerd-failover-tests/templates/tests/curl-pod.yaml:21]: can you use `for i in $(seq 10)` here?

**Review by adleong — APPROVED** [review]

**Pull request merged**